ansible_collections/arista/avd/plugins/action/validate_and_template.py

```diff
@@ -42,8 +42,21 @@ def run(self, tmp=None, task_vars=None):
 
         # Build data from hostvars and role default vars
         hostname = task_vars["inventory_hostname"]
-        self.data = self._templar.template(self._task._role.get_default_vars())
-        self.data.update(task_vars["hostvars"].get(hostname))
+
+        # Read ansible variables and perform templating to support inline jinja2
+        for var in task_vars:
+            if str(var).startswith(("ansible", "molecule", "hostvars", "vars", "avd_switch_facts")):
+                continue
+            if self._templar.is_template(task_vars[var]):
+                # Var contains a jinja2 template.
+                try:
+                    task_vars[var] = self._templar.template(task_vars[var], fail_on_undefined=False)
+                except Exception as e:
+                    raise AnsibleActionFail(f"Exception during templating of task_var '{var}'") from e
+
+        if not isinstance(task_vars, dict):
+            # Corner case for ansible-test where the passed task_vars is a nested chain-map
+            task_vars = dict(task_vars)
```

Comment:
Should we not rather pass the correct type from ansible-test?

Reply:
we don't control that